File: C-Language/mergesort.c
#include <stdio.h>

void merge(int list[], int sorted[], int left, int mid, int right) {
  int i = left, j = mid + 1, k = left;

  while (i <= mid && j <= right) {
    if (list[i] <= list[j])
      sorted[k++] = list[i++];
    else
      sorted[k++] = list[j++];
  }

  while (i <= mid)
    sorted[k++] = list[i++];

  while (j <= right)
    sorted[k++] = list[j++];

  for (i = left; i <= right; i++)
    list[i] = sorted[i];
}

void mergeSort(int list[], int sorted[], int left, int right) {
  if (left < right) {
    int mid = (left + right) / 2;
    mergeSort(list, sorted, left, mid);
    mergeSort(list, sorted, mid + 1, right);
    merge(list, sorted, left, mid, right);
  }
}

int main() {
  int list[] = {34, 7, 23, 32, 5, 62};
  int sorted[6];
  int size = sizeof(list) / sizeof(list[0]);

  mergeSort(list, sorted, 0, size - 1);

  for (int i = 0; i < size; i++)
    printf("%d ", list[i]);

  return 0;
}

File: Python/mergesort.py
def merge(list, left, mid, right):
    n1 = mid - left + 1
    n2 = right - mid

    L = [0] * n1
    R = [0] * n2

    for i in range(0, n1):
        L[i] = list[left + i]

    for j in range(0, n2):
        R[j] = list[mid + 1 + j]

    i = 0
    j = 0
    k = left

    while i < n1 and j < n2:
        if L[i] <= R[j]:
            list[k] = L[i]
            i += 1
        else:
            list[k] = R[j]
            j += 1
        k += 1

    while i < n1:
        list[k] = L[i]
        i += 1
        k += 1

    while j < n2:
        list[k] = R[j]
        j += 1
        k += 1


def mergeSort(list, left, right):
    if left < right:
        mid = (left + right) // 2

        mergeSort(list, left, mid)
        mergeSort(list, mid + 1, right)
        merge(list, left, mid, right)


list = [34, 7, 23, 32, 5, 62]
mergeSort(list, 0, len(list) - 1)
print("Sorted array is:", list)
